mlrun/datastore/targets.py

```diff
@@ -1568,6 +1561,44 @@ def prepare_spark_df(self, df, key_columns, timestamp_key=None, spark_options=No
 
         return df
 
+    def add_writer_step(
+        self,
+        graph,
+        after,
+        features,
+        key_columns=None,
+        timestamp_key=None,
+        featureset_status=None,
+    ):
+        key_columns = list(key_columns.keys())
+        table = self._resource.uri
+        column_list = self._get_column_list(
+            features=features,
+            timestamp_key=None,
+            key_columns=key_columns,
+            with_type=True,
+        )
+        if not self.columns:
+            aggregate_features = (
+                [key for key, feature in features.items() if feature.aggregate]
+                if features
+                else []
+            )
+            column_list = [
+                col for col in column_list if col[0] not in aggregate_features
+            ]
```

Comment:
Looks like this duplicated code can be moved to the base class to avoid duplicating these lines.

Reply:
ok